workorders.lic

```diff
@@ -258,32 +260,40 @@ class WorkOrders
             bput('combine', 'combine')
           end
         else
-          dispose("#{info['stock-name']} stone") if count > 0 && spare_stock
           order_item(info['stock-room'], info[stock_number])
-          fput('tap my deed')
+          if @engineering_room
+            fput('tap my deed')
+            material_volume = info['stock-volume']
+          else
+            material_volume = 0
+          end
         end
         stow_hands
       end
 
       if !bone_carving
-        rock_result = bput("get #{info['stock-name']} rock", 'You get', 'What were', 'You are not strong')
-        find_shaping_room(@hometown, @engineering_room) unless rock_result == 'You are not strong'
+        rock_result = bput("get #{info['stock-name']} #{material_noun[material_volume]}", 'You get', 'What were', 'You are not strong', 'You pick up', 'but can\'t quite lift it')
+        find_shaping_room(@hometown, @engineering_room) unless rock_result =~ /You are not strong|but can\'t quite lift it/i
       else
         find_shaping_room(@hometown, @engineering_room)
       end
+      bput('swap', 'You move') if right_hand =~ /#{@noun}/i
 
-      wait_for_script_to_complete('carve', [recipe['chapter'], recipe['name'], info[stock_name], bone_carving ? 'stack' : 'rock', recipe['noun']])
+      wait_for_script_to_complete('carve', [recipe['chapter'], recipe['name'], info[stock_name], bone_carving ? 'stack' : material_noun[material_volume], recipe['noun']])
+      
+      material_volume = info['stock-volume'] if material_volume == 0
+      material_volume -= recipe['volume']
```

Comment:
Why subtract `recipe['volume']` here just to add it again on line 296?

Reply:
its not needed actually...the -=recipe['volume'] takes care of the issue.